### PR TITLE
Added programmatic description fields

### PR DESCRIPTION
**Pull request by samshuster**
### Summary of Changes

https://github.com/lyft/amundsen/issues/147

I added the inclusion of programmatic descriptions in the returned table metadata object.

### Tests

I modified the proxy-neo4j tests to test for programmatic description.

### Documentation

None added.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

File: metadata_service/proxy/neo4j_proxy.py [hunk 1]
```diff
@@ -236,7 +241,19 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
             src = Source(source_type=table_records['src']['source_type'],
                          source=table_records['src']['source'])
 
-        return wmk_results, table_writer, timestamp_value, owner_record, tags, src, badges
+        prog_descriptions = self._extract_programmatic_descriptions_from_query(table_records)
```
Comment (jinhyukchang) [inline metadata_service/proxy/neo4j_proxy.py]: Suggestion: How about passing `prog_descriptions` , an output from `table_records.get('prog_descriptions', [])`, as a method parameter? Just to decouple `table_records` structure from `_extract_programmatic_descriptions_from_query` method.

e.g: _extract_programmatic_descriptions_from_query(table_records.get('prog_descriptions', []))

File: metadata_service/proxy/neo4j_proxy.py [hunk 2]
```diff
@@ -236,7 +241,19 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
...
+
+        return wmk_results, table_writer, timestamp_value, owner_record, tags, src, badges, prog_descriptions
+
+    def _extract_programmatic_descriptions_from_query(self, table_records: dict) -> list:
+        prog_descriptions = []
+        for prog_description in table_records.get('prog_descriptions', []):
+            LOGGER.info(prog_description)
```
Comment (jinhyukchang) [inline metadata_service/proxy/neo4j_proxy.py]: Could you remove this line?

File: tests/unit/api/table/test_table_detail_api.py [hunk 3]
```diff
@@ -63,7 +65,7 @@ def test_should_get_column_details(self) -> None:
         self.mock_proxy.get_table.return_value = QUERY_RESPONSE
 
         response = self.app.test_client().get(f'/table/{TABLE_URI}')
-
+        print(response.json)
```
Comment (jinhyukchang) [inline tests/unit/api/table/test_table_detail_api.py]: Could you remove this line as well?

File: metadata_service/proxy/neo4j_proxy.py [hunk 4]
```diff
@@ -236,7 +241,19 @@ def _exec_table_query(self, table_uri: str) -> Tuple:
...
+            source = prog_description['description_source']
```
Comment (jinhyukchang) [inline metadata_service/proxy/neo4j_proxy.py]: Just to confirm if `description_source` is guaranteed to be there.

File: metadata_service/proxy/neo4j_proxy.py (hunk 2 above)
Comment (samshuster) [reply metadata_service/proxy/neo4j_proxy.py]: done

File: tests/unit/api/table/test_table_detail_api.py (hunk 3 above)
Comment (samshuster) [reply tests/unit/api/table/test_table_detail_api.py]: done

File: metadata_service/proxy/neo4j_proxy.py (hunk 1 above)
Comment (samshuster) [reply metadata_service/proxy/neo4j_proxy.py]: good idea

File: metadata_service/proxy/neo4j_proxy.py (hunk 4 above)
Comment (samshuster) [reply metadata_service/proxy/neo4j_proxy.py]: yes if it isn't there then something bad happened with data builder. But... better to be safe here. Thanks for pointing this out.

**Comment by samshuster** [comment]
ok i think i addressed all of your concerns @jinhyukchang! also revved minor version

**Pull request merged**